### PR TITLE
Add mentions

### PR DESCRIPTION
**Pull request by metoou**
Add mentions for: csharp-mvc, csharp, java, php, python and ruby projects

### CODE REVIEW

**Review by aleksanderfedorov — APPROVED** [review]

File: web/documentserver-example/nodejs/app.js [hunk 1]
```diff
@@ -34,6 +34,7 @@ const docManager = require("./helpers/docManager");
 const documentService = require("./helpers/documentService");
 const fileUtility = require("./helpers/fileUtility");
 const users = require("./helpers/users");
+const { getUsersForMentions } = require("./helpers/users");
```
Comment (LinneyS) [inline web/documentserver-example/nodejs/app.js:37]: this line is not required

File: web/documentserver-example/nodejs/views/editor.ejs [hunk 2]
```diff
@@ -149,6 +162,8 @@
                         "onRequestInsertImage": onRequestInsertImage,
                         "onRequestCompareFile": onRequestCompareFile,
                         "onRequestMailMergeRecipients": onRequestMailMergeRecipients,
+                        "onRequestUsers": onRequestUsers,
+                        "onRequestSendNotify": onRequestSendNotify,
```
Comment (LinneyS) [inline web/documentserver-example/nodejs/views/editor.ejs:166]: Do not send this functions for anonymous

File: web/documentserver-example/csharp/DocEditor.aspx.cs [hunk 3]
```diff
@@ -66,6 +66,7 @@ protected string DocServiceApiUri
         protected string InsertImageConfig { get; private set; }
         protected string compareFileData { get; private set; }
         protected string dataMailMergeRecipients { get; private set; }
+        protected string users { get; private set; }
```
Comment (LinneyS) [inline web/documentserver-example/csharp/DocEditor.aspx.cs:69]: rename to `usersForMentions`

File: web/documentserver-example/java/src/main/java/controllers/EditorServlet.java [hunk 4]
```diff
@@ -88,6 +89,9 @@ protected void processRequest(HttpServletRequest request, HttpServletResponse re
         dataMailMergeRecipients.put("fileType", "csv");
         dataMailMergeRecipients.put("url", DocumentManager.GetServerUrl(true) + "/IndexServlet?type=csv");
 
+        // users data for mentions
+        List<Map<String, Object>> usersData = !user.id.equals("uid-0") ? Users.getUsersForMentions(user.id) : null;
```
Comment (LinneyS) [inline web/documentserver-example/java/src/main/java/controllers/EditorServlet.java:93]: rename to `usersForMentions`

File: web/documentserver-example/nodejs/views/editor.ejs (hunk 2 above)
Comment (LinneyS) [reply web/documentserver-example/nodejs/views/editor.ejs:166]: on all lang change the same

File: web/documentserver-example/php/doceditor.php [hunk 5]
```diff
@@ -135,6 +135,9 @@
         "fileType" =>"csv",
         "url" => serverPath(true) . "/webeditor-ajax.php?type=csv"
     ];
+    
+    // users data for mentions
+    $users = $user->id != "uid-0" ? getUsersForMentions($user->id) : null;
```
Comment (LinneyS) [inline web/documentserver-example/php/doceditor.php:140]: rename to `usersForMentions`

File: web/documentserver-example/python/src/views/actions.py [hunk 6]
```diff
@@ -222,6 +222,9 @@ def edit(request):
         'url': docManager.getServerUrl(True, request) + 'csv'
     }
 
+    # users data for mentions
+    usersData = users.getUsersForMentions(user.id) if user.id !='uid-0' else None
```
Comment (LinneyS) [inline web/documentserver-example/python/src/views/actions.py:226]: rename to `usersForMentions`

File: web/documentserver-example/python/src/views/actions.py [hunk 7]
```diff
@@ -238,7 +241,8 @@ def edit(request):
         'apiUrl': config.DOC_SERV_SITE_URL + config.DOC_SERV_API_URL,  # the absolute URL to the api
         'dataInsertImage': json.dumps(dataInsertImage)[1 : len(json.dumps(dataInsertImage)) - 1],  # the image which will be inserted into the document
         'dataCompareFile': dataCompareFile,  # document which will be compared with the current document
-        'dataMailMergeRecipients': json.dumps(dataMailMergeRecipients)  # recipient data for mail merging
+        'dataMailMergeRecipients': json.dumps(dataMailMergeRecipients),  # recipient data for mail merging
+        'users': json.dumps(usersData)
```
Comment (LinneyS) [inline web/documentserver-example/python/src/views/actions.py:245]: rename to `usersForMentions`

**Review by LinneyS — CHANGES_REQUESTED** [review]

**Pull request merged**